server/src/main/java/org/opensearch/search/profile/query/InternalProfileCollectorManager.java

```diff
@@ -46,9 +49,35 @@ public InternalProfileCollectorManager(
     ) {
         this.manager = manager;
         this.reason = reason;
+        this.collectorManagerName = deriveCollectorManagerName(manager);
         this.children = children;
     }
 
+    /**
+     * Creates a human-friendly representation of the CollectorManager name.
+     *
+     * @param manager The CollectorManager to derive a name from
+     * @return  A (hopefully) prettier name
+     */
+    private String deriveCollectorManagerName(CollectorManager<? extends Collector, ReduceableSearchResult> manager) {
+        String name = manager.getClass().getSimpleName();
+        if (name.equals("")) {
+            name = manager.getClass().getEnclosingClass().getSimpleName();
+        }
+
+        // Include the user-defined agg name
+        String userDefinedAggName = null;
+        if (reason.equals(CollectorResult.REASON_AGGREGATION)) {
+            userDefinedAggName = ((NonGlobalAggCollectorManager) manager).getName();
```

Comment:
Should we make it into `AggregationCollectorManager::getName()` (as we did for reason) ?

Reply:
Hey @reta, It has been addressed now.